### ISSUE TITLE
[2021] Towards Quantifiable Dialogue Coherence Evaluation

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#142]
## 0. 論文

| タイトル | Towards Quantifiable Dialogue Coherence Evaluation |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2106.00507 https://github.com/James-Yip/QuantiDCE |
| 著者    | Zheng Ye, Liucun Lu, Lishan Huang, Liang Lin, Xiaodan Liang |
| 投稿日付 | ACL2021 |

## 1. どんなもの？

![スクリーンショット 2021-08-14 16 41 22](https://user-images.githubusercontent.com/1255485/129439052-52444b9e-64f6-46e5-b6e7-ec3864032bfd.png)

対話システムの自動評価指標において、NNによる評価モデルが近年提案されているが、既存研究では coherent / incoherent を2値で考えて評価算出モデルを作成しているものが多い中、coherent をより細かく定量化しそれを使用する学習方法  QuantiDCE (Quantifiable Dialogue Coherence Evaluation) を提案。
学習方法を工夫することにより少量のラベル付きデータで人手評価による複数段階の coherence スコアを学習させ、実験にて SoTA な自動評価指標よりも高い人間との相関を得た。

## 2. 先行研究と比べてどこがすごい？

- coherence を2値ではなく定量化して学習させた対話の自動評価指標は初

## 3. 技術や手法のキモはどこ？

![スクリーンショット 2021-08-14 16 41 35](https://user-images.githubusercontent.com/1255485/129439054-a8e1b336-aa4e-41e9-9180-fceea741b35c.png)

- モデル自体はシンプルで、BERT encoder の上に3層の MLP を加え、スコア (scalar) を算出
- 学習は2段階: pre-training, fine-tuning with distillation
- pre-training: multi-level ranking (MLR) pre-training
    - 3つのロスで学習: separation loss, compactness loss, ordering loss
    - separation loss
        - 各 coherence level ごとにスコアの中心を計算し、異なる coherence level の中心点が離れるように学習する
        - ![スクリーンショット 2021-08-14 19 27 26](https://user-images.githubusercontent.com/1255485/129443369-48144f0a-2a9d-4555-a109-0b43ba7156ec.png)
        - ここで d は L1 Loss, lambda は2つの中心間の lower bounds, w は coherence level の差
    - compactness loss
        - あるサンプルのスコアが、その coherence level 中心点と近づくように学習する
        - ![スクリーンショット 2021-08-14 19 29 55](https://user-images.githubusercontent.com/1255485/129443428-d3e8ea2b-65b1-42ef-93d2-9e155f0dcb88.png)
    - ordering loss
        - coherence level ごとのスコアの中心点の order が正しい順序で並ぶように学習
        - ![スクリーンショット 2021-08-14 19 31 44](https://user-images.githubusercontent.com/1255485/129443464-9f39a496-ec3e-48c7-bcbc-1b2e9fcc1b4f.png)
- fine-tuning: Knowledge Distillation (KD) Fine-Tuning
    - 少量の人手ラベル付きデータで定量化された coherent を学習するが、少量のためそのままでは過学習する
    - pre-training したモデルを使って、TinyBERT の方法で Distillation しながら学習する
    - 単純な MSE と、KD による regularization の2つで効率的に学習

## 4. どうやって有効だと検証した？

人手評価との相関(Pearson, Spearman, Kendall) を計測。

- ベンチマークデータセット
    - GRADE で使用された large-scale human judgement benchmark (DailyDialog/ConvAI2/EmpatheticDialogues の test data から作成したもの)
    - 返答は retrieval-base と generation-base 両方使用
- ベースラインメトリック
    - lexical word-overlap ベースのもの: BLEU, ROUGE, METEOR
    - semantic 考慮するもの: BERTScore, ADEM, BERT-RUBER, BLEURT, GRADE
- 学習設定
    - 学習データは pre-training に DailyDialog++, fine-tuning に DailyDialogEVAL
    - coherent の粒度Lには `L = 3` を使用 (DailyDialog++ は random/adversarial/gold の人手データがあるので、それぞれを coherent 1~3 として使用)
- 結果:
    - SoTA である GRADE を上回るスコア
    - ![スクリーンショット 2021-08-14 19 10 26](https://user-images.githubusercontent.com/1255485/129442671-a8f7ce94-0284-4a15-93ca-5335ae331cc9.png)
    - いくつかの ablation study より、pre-training, fine-tuning それぞれのロスが効果ありとのこと

## 5. 議論はある？

- [IMO] アイディアはストレートだけど、結局事前学習時の教師データは勝手に coherent を与えているが良いのだろうか?あくまで pre-training なのでそこまで影響はないかもしれないが
    - 例えば5段階にしたとき random が1, adversarial が4, gold が5の場合と、random が1, adversarial が2, gold が5の場合で全然違う

## 6. 次に読むべき論文は？

- GRADE: Automatic Graph-Enhanced Coherence Metric for Evaluating Open-Domain Dialogue Systems
    - https://github.com/cfiken/paper-reading/issues/110
    - ベースラインモデルとして挙げられている、SoTA手法
